Windows error in test_array_core.py::test_memmap

An error trying to close a file while another process has access to it.
```python
_________________________________ test_memmap _________________________________

    def test_memmap():
        with tmpfile('npy') as fn_1:
            with tmpfile('npy') as fn_2:
                x = da.arange(100, chunks=15)
                target = np.memmap(fn_1, shape=x.shape, mode='w+', dtype=x.dtype)

                x.store(target)

                assert eq(target, x)

                np.save(fn_2, target)

>               assert eq(np.load(fn_2, mmap_mode='r'), x)

tests\test_array_core.py:1439:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\..\..\..\anaconda\envs\dask\lib\contextlib.py:24: in __exit__
    self.gen.next()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

extension = '.npy', dir = None

    @contextmanager
    def tmpfile(extension='', dir=None):
        extension = '.' + extension.lstrip('.')
        handle, filename = tempfile.mkstemp(extension, dir=dir)
        os.close(handle)
        os.remove(filename)

        try:
            yield filename
        finally:
            if os.path.exists(filename):
                if os.path.isdir(filename):
                    shutil.rmtree(filename)
                else:
>                   os.remove(filename)
E                   WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\us
ers\\ieuser\\appdata\\local\\temp\\tmpymijv3.npy'
```

--- a/dask/array/tests/test_array_core.py
+++ b/dask/array/tests/test_array_core.py
@@ -1427,16 +1427,19 @@ def test_cov():
 def test_memmap():
     with tmpfile('npy') as fn_1:
         with tmpfile('npy') as fn_2:
-            x = da.arange(100, chunks=15)
-            target = np.memmap(fn_1, shape=x.shape, mode='w+', dtype=x.dtype)
+            try:
+                x = da.arange(100, chunks=15)
+                target = np.memmap(fn_1, shape=x.shape, mode='w+', dtype=x.dtype)
 
-            x.store(target)
+                x.store(target)
 
-            assert eq(target, x)
+                assert eq(target, x)
 
-            np.save(fn_2, target)
+                np.save(fn_2, target)
 
-            assert eq(np.load(fn_2, mmap_mode='r'), x)
+                assert eq(np.load(fn_2, mmap_mode='r'), x)
+            finally:
+                target._mmap.close()
 
 
 def test_to_npy_stack():

